src/invariant/EventsAreConsistentWithEntryDiffs.cpp

```diff
@@ -22,14 +22,18 @@ static const CxxI128 I128ZERO{0, 0};
 
 struct AggregatedEvents
 {
-    mutable UnorderedMap<SCAddress, UnorderedMap<Asset, CxxI128>> mEventAmounts;
+    UnorderedMap<SCAddress, UnorderedMap<Asset, CxxI128>> mEventAmounts;
+    UnorderedMap<SCAddress, UnorderedMap<Asset, std::optional<bool>>>
+        mIsAuthorized;
     UnorderedMap<Hash, Asset> mStellarAssetContractIDs;
```

Comment:
Since previously you mentioned using `mutable` is in order to pass `AggregatedEvents` as `const` so that `mStellarAssetContractIDs` can stay immutable. Perhaps it's better to just split out the `mStellarAssetContractIDs`, since it's just a cache and can live outside of `AggregatedEvents` and stay constant.

Reply:
Good point. Updated